### ISSUE TITLE
Error on OpenAFS FATAL:   container creation failed: mount /proc/self/fd/3->/opt/singularity-3.8.4/var/singularity/mnt/session/rootfs 

### ISSUE THREAD

**Issue by angelinux** [sylabs/singularity#423]
**Version of Singularity**
3.8.4

**Describe the bug**
Unable to run  Singularity on OpenAFS filesystem

**To Reproduce**
- Install openafs client 1.6
- on an AFS filesystem run `singularity shell pytorch.simg` where pytorch.simg is an image containing only pytorch libraries 1.6.0
- the result is `FATAL:   container creation failed: mount /proc/self/fd/3->/opt/singularity-3.8.4/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/3: failed to mount squashfs filesystem: input/output error` 

**Expected behavior**
We expect a shell where we can write `python -E` and then at the python prompt: `import torch; print(torch.__version__)`

**OS / Linux Distribution**
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Installation Method**
Source installation with prefix `/opt/singularity-3.8.4/`

**Additional context**
This problem is related to the OpenAFS filesystem client 1.6/1.8

**Comment by dtrudg** [comment]
Please provide full debug output from `singularity -d ....` so we can see exactly what is getting mounted where.

Is just the image on the AFS filesystem, the singularity install?

**Comment by angelinux** [comment]
This is the output of `singularity -d` :

```
/opt/singularity-3.8.4/bin/singularity -d shell pytorch.simg 
DEBUG   [U=26806,P=49390]  persistentPreRun()            Singularity version: 3.8.4
DEBUG   [U=26806,P=49390]  persistentPreRun()            Parsing configuration file /opt/singularity-3.8.4/etc/singularity/singularity.conf
DEBUG   [U=26806,P=49390]  handleConfDir()               /afs/enea.it/bri/user/mariano/.singularity already exists. Not creating.
DEBUG   [U=26806,P=49390]  execStarter()                 Saving umask 0022 for propagation into container
DEBUG   [U=26806,P=49390]  execStarter()                 Checking for encrypted system partition
DEBUG   [U=26806,P=49390]  Init()                        Image format detection
DEBUG   [U=26806,P=49390]  Init()                        Check for sandbox image format
DEBUG   [U=26806,P=49390]  Init()                        sandbox format initializer returned: not a directory image
DEBUG   [U=26806,P=49390]  Init()                        Check for sif image format
DEBUG   [U=26806,P=49390]  Init()                        sif image format detected
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding MKLROOT environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding MANPATH environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding XDG_SESSION_ID environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding HOSTNAME environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SELINUX_ROLE_REQUESTED environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding IPPROOT environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding INTEL_LICENSE_FILE environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding TERM environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SHELL environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding HISTSIZE environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SSH_CLIENT environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SELINUX_USE_CURRENT_RANGE environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding FPATH environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SSH_TTY environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding USER environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding LS_COLORS environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding CPATH environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding NLSPATH environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding MAIL environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding ENEAGRID_CLUSTER_ID environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding PWD environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding _LMFILES_ environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding LANG environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding MODULEPATH environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding LOADEDMODULES environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding PS1 environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SELINUX_LEVEL_REQUESTED environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding HISTCONTROL environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding KRB5CCNAME environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SHLVL environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding MODULERCFILE environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding LOGNAME environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding XDG_DATA_DIRS environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding SSH_CONNECTION environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding MODULESHOME environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding LESSOPEN environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding XDG_RUNTIME_DIR environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding INCLUDE environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding BASH_FUNC_module() environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding _ environment variable
DEBUG   [U=26806,P=49390]  SetContainerEnv()             Forwarding USER_PATH environment variable
VERBOSE [U=26806,P=49390]  SetContainerEnv()             Setting HOME=/afs/enea.it/bri/user/mariano
VERBOSE [U=26806,P=49390]  SetContainerEnv()             Setting PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
DEBUG   [U=26806,P=49390]  init()                        Use starter binary /opt/singularity-3.8.4/libexec/singularity/bin/starter-suid
DEBUG   [U=26806,P=49390]  Exec()                        Setting GOGC=off for starter
VERBOSE [U=0,P=49390]      print()                       Set messagelevel to: 5
VERBOSE [U=0,P=49390]      init()                        Starter initialization
DEBUG   [U=0,P=49390]      load_overlay_module()         Trying to load overlay kernel module
DEBUG   [U=0,P=49390]      load_overlay_module()         Overlay seems supported by the kernel
VERBOSE [U=0,P=49390]      is_suid()                     Check if we are running as setuid
VERBOSE [U=0,P=49390]      priv_drop()                   Drop root privileges
DEBUG   [U=26806,P=49390]  read_engine_config()          Read engine configuration
DEBUG   [U=26806,P=49390]  cleanenv()                    Clearing env var LOAD_OVERLAY_MODULE=1
DEBUG   [U=26806,P=49390]  cleanenv()                    Keeping env var SINGULARITY_MESSAGELEVEL=5
DEBUG   [U=26806,P=49390]  cleanenv()                    Clearing env var ENGINE_CONFIG1={"engineName":"singularity","containerID":"","engineConfig":{"jsonConfig":{"unixSocketPair":[0,0],"image":"/afs/enea.it/bri/user/mariano/pytorch.simg","homedir":"/afs/enea.it/bri/user/mariano","homeDest":"/afs/enea.it/bri/user/mariano","network":"bridge","cwd":"/afs/enea.it/bri/user/mariano","configurationFile":"/opt/singularity-3.8.4/etc/singularity/singularity.conf","restoreUmask":true,"umask":18},"ociConfig":{"ociVersion":"","process":{"user":{"uid":0,"gid"(TRUNCATED...)DEBUG   [U=26806,P=49390]  cleanenv()                    Clearing env var ENGINE_CONFIG_CHUNKS=1
DEBUG   [U=26806,P=49390]  cleanenv()                    Keeping env var GOGC=off
DEBUG   [U=26806,P=49390]  init()                        Wait completion of stage1
VERBOSE [U=26806,P=49405]  priv_drop()                   Drop root privileges permanently
DEBUG   [U=26806,P=49405]  set_parent_death_signal()     Set parent death signal to 9
VERBOSE [U=26806,P=49405]  init()                        Spawn stage 1
DEBUG   [U=26806,P=49405]  startup()                     singularity runtime engine selected
VERBOSE [U=26806,P=49405]  startup()                     Execute stage 1
DEBUG   [U=26806,P=49405]  StageOne()                    Entering stage 1
DEBUG   [U=26806,P=49405]  prepareUserCaps()             Ignoring group 1099606120: group: unknown groupid 1099606120
DEBUG   [U=26806,P=49405]  prepareAutofs()               Found "/proc/sys/fs/binfmt_misc" as autofs mount point
DEBUG   [U=26806,P=49405]  prepareAutofs()               Could not keep file descriptor for bind path /etc/localtime: no mount point
DEBUG   [U=26806,P=49405]  prepareAutofs()               Could not keep file descriptor for bind path /etc/hosts: no mount point
DEBUG   [U=26806,P=49405]  prepareAutofs()               Could not keep file descriptor for home directory /afs/enea.it/bri/user/mariano: no mount point
DEBUG   [U=26806,P=49405]  prepareAutofs()               Could not keep file descriptor for current working directory /afs/enea.it/bri/user/mariano: no mount point
DEBUG   [U=26806,P=49405]  Init()                        Image format detection
DEBUG   [U=26806,P=49405]  Init()                        Check for sandbox image format
DEBUG   [U=26806,P=49405]  Init()                        sandbox format initializer returned: not a directory image
DEBUG   [U=26806,P=49405]  Init()                        Check for sif image format
DEBUG   [U=26806,P=49405]  Init()                        sif image format detected
DEBUG   [U=26806,P=49405]  setSessionLayer()             Overlay seems supported and allowed by kernel
DEBUG   [U=26806,P=49405]  setSessionLayer()             Attempting to use overlayfs (enable overlay = try)
VERBOSE [U=26806,P=49390]  wait_child()                  stage 1 exited with status 0
DEBUG   [U=26806,P=49390]  cleanup_fd()                  Setting FD_CLOEXEC on starter fd 3
DEBUG   [U=26806,P=49390]  cleanup_fd()                  Close file descriptor 4
DEBUG   [U=26806,P=49390]  cleanup_fd()                  Close file descriptor 5
DEBUG   [U=26806,P=49390]  cleanup_fd()                  Close file descriptor 6
DEBUG   [U=26806,P=49390]  init()                        Set child signal mask
DEBUG   [U=26806,P=49390]  init()                        Create socketpair for master communication channel
DEBUG   [U=26806,P=49390]  init()                        Create RPC socketpair for communication between stage 2 and RPC server
VERBOSE [U=26806,P=49390]  priv_escalate()               Get root privileges
VERBOSE [U=0,P=49390]      priv_escalate()               Change filesystem uid to 26806
VERBOSE [U=0,P=49390]      init()                        Spawn master process
DEBUG   [U=0,P=49411]      set_parent_death_signal()     Set parent death signal to 9
VERBOSE [U=0,P=49411]      create_namespace()            Create mount namespace
VERBOSE [U=0,P=49390]      enter_namespace()             Entering in mount namespace
DEBUG   [U=0,P=49390]      enter_namespace()             Opening namespace file ns/mnt
VERBOSE [U=0,P=49411]      create_namespace()            Create mount namespace
DEBUG   [U=0,P=49390]      set_master_privileges()       Set master privileges
DEBUG   [U=0,P=49390]      apply_privileges()            Effective capabilities:   0x00000000000000c0
DEBUG   [U=0,P=49390]      apply_privileges()            Permitted capabilities:   0x0000001fffffffff
DEBUG   [U=0,P=49390]      apply_privileges()            Bounding capabilities:    0x0000001fffffffff
DEBUG   [U=0,P=49390]      apply_privileges()            Inheritable capabilities: 0x0000001fffffffff
DEBUG   [U=0,P=49390]      apply_privileges()            Ambient capabilities:     0x0000000000000000
DEBUG   [U=0,P=49390]      apply_privileges()            Set user ID to 26806
DEBUG   [U=0,P=49412]      set_rpc_privileges()          Set RPC privileges
DEBUG   [U=0,P=49412]      apply_privileges()            Effective capabilities:   0x0000000000200000
DEBUG   [U=0,P=49412]      apply_privileges()            Permitted capabilities:   0x0000001fffffffff
DEBUG   [U=0,P=49412]      apply_privileges()            Bounding capabilities:    0x0000000008204000
DEBUG   [U=0,P=49412]      apply_privileges()            Inheritable capabilities: 0x0000000000000000
DEBUG   [U=0,P=49412]      apply_privileges()            Ambient capabilities:     0x0000000000000000
DEBUG   [U=0,P=49412]      apply_privileges()            Set user ID to 26806
DEBUG   [U=26806,P=49412]  set_parent_death_signal()     Set parent death signal to 9
VERBOSE [U=26806,P=49412]  init()                        Spawn RPC server
DEBUG   [U=26806,P=49390]  startup()                     singularity runtime engine selected
VERBOSE [U=26806,P=49390]  startup()                     Execute master process
DEBUG   [U=26806,P=49412]  startup()                     singularity runtime engine selected
VERBOSE [U=26806,P=49412]  startup()                     Serve RPC requests
DEBUG   [U=26806,P=49390]  setupSessionLayout()          Using Layer system: overlay
DEBUG   [U=26806,P=49390]  setupOverlayLayout()          Creating overlay SESSIONDIR layout
DEBUG   [U=26806,P=49390]  addRootfsMount()              Mount rootfs in read-only mode
DEBUG   [U=26806,P=49390]  addRootfsMount()              Image type is 4096
DEBUG   [U=26806,P=49390]  addRootfsMount()              Mounting block [squashfs] image: /afs/enea.it/bri/user/mariano/pytorch.simg
DEBUG   [U=26806,P=49390]  addKernelMount()              Checking configuration file for 'mount proc'
DEBUG   [U=26806,P=49390]  addKernelMount()              Adding proc to mount list
VERBOSE [U=26806,P=49390]  addKernelMount()              Default mount: /proc:/proc
DEBUG   [U=26806,P=49390]  addKernelMount()              Checking configuration file for 'mount sys'
DEBUG   [U=26806,P=49390]  addKernelMount()              Adding sysfs to mount list
VERBOSE [U=26806,P=49390]  addKernelMount()              Default mount: /sys:/sys
DEBUG   [U=26806,P=49390]  addDevMount()                 Checking configuration file for 'mount dev'
DEBUG   [U=26806,P=49390]  addDevMount()                 Adding dev to mount list
VERBOSE [U=26806,P=49390]  addDevMount()                 Default mount: /dev:/dev
DEBUG   [U=26806,P=49390]  addHostMount()                Not mounting host file systems per configuration
VERBOSE [U=26806,P=49390]  addBindsMount()               Found 'bind path' = /etc/localtime, /etc/localtime
VERBOSE [U=26806,P=49390]  addBindsMount()               Found 'bind path' = /etc/hosts, /etc/hosts
DEBUG   [U=26806,P=49390]  addHomeStagingDir()           Staging home directory (/afs/enea.it/bri/user/mariano) at /opt/singularity-3.8.4/var/singularity/mnt/session/afs/enea.it/bri/user/mariano
DEBUG   [U=26806,P=49390]  addHomeMount()                Adding home directory mount [/opt/singularity-3.8.4/var/singularity/mnt/session/afs/enea.it/bri/user/mariano:/afs/enea.it/bri/user/mariano] to list using layer: overlay
DEBUG   [U=26806,P=49390]  addTmpMount()                 Checking for 'mount tmp' in configuration file
VERBOSE [U=26806,P=49390]  addTmpMount()                 Default mount: /tmp:/tmp
VERBOSE [U=26806,P=49390]  addTmpMount()                 Default mount: /var/tmp:/var/tmp
DEBUG   [U=26806,P=49390]  addScratchMount()             Not mounting scratch directory: Not requested
DEBUG   [U=26806,P=49390]  addLibsMount()                Checking for 'user bind control' in configuration file
DEBUG   [U=26806,P=49390]  addFilesMount()               Checking for 'user bind control' in configuration file
DEBUG   [U=26806,P=49390]  addResolvConfMount()          Adding /etc/resolv.conf to mount list
VERBOSE [U=26806,P=49390]  addResolvConfMount()          Default mount: /etc/resolv.conf:/etc/resolv.conf
DEBUG   [U=26806,P=49390]  addHostnameMount()            Skipping hostname mount, not virtualizing UTS namespace on user request
DEBUG   [U=26806,P=49390]  create()                      Mount all
DEBUG   [U=26806,P=49390]  mountGeneric()                Mounting tmpfs to /opt/singularity-3.8.4/var/singularity/mnt/session
DEBUG   [U=26806,P=49390]  mountImage()                  Mounting loop device /dev/loop0 to /opt/singularity-3.8.4/var/singularity/mnt/session/rootfs of type squashfs
FATAL   [U=26806,P=49390]  Master()                      container creation failed: mount /proc/self/fd/3->/opt/singularity-3.8.4/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/3: failed to mount squashfs filesystem: input/output error
```

Singularity is installed on the server filesystem, while the `pytorch.simg` is located in the `/afs/enea.it/bri/user/mariano` filesystem, based on OpenAFS.
Thanks,
Angelo

**Comment by dtrudg** [comment]
Thanks for the further information. Nothing untoward appears here in the logs except the final error, and running the image off of an afs filesystem like this is something we know of other users doing.

The `input/output error` would suggest there is something amiss with the image, or the AFS fs / squashfs operations themselves rather than anything Singularity is doing itself.

I'm afraid that further debugging will need some more investigation:

1) Are there any errors in `dmesg` from this failure

2) If you force extraction of the image to a temporary dir with the `-u` user namespace option to `singularity shell`, does the image run?

If you can't use a usernamespace, try to use `singularity sif dump` and then `unsquashfs` on the resulting image. Or perform a sandbox build, e.g....

```
$ cd /tmp
$ singularity build -s 
``` 

3. If you use `mksquashfs` to create an arbitrary squashfs image on the AFS filesystem (`mksquashfs mydir test.squashfs`), can you then successfully use it from AFS, completely independently of Singularity e.g. `mount -o loop test.squashfs /tmp/mountdir`

**Comment by angelinux** [comment]
1. This is the output of dmesg:

```
[148588.504007] loop: module loaded
[148588.517973] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[148588.564967] Buffer I/O error on dev loop0, logical block 3838912, async page read
[148588.576241] Buffer I/O error on dev loop0, logical block 3838913, async page read
[148588.587570] Buffer I/O error on dev loop0, logical block 3838914, async page read
[148588.598943] Buffer I/O error on dev loop0, logical block 3838915, async page read
[148588.610372] SQUASHFS error: squashfs_read_data failed to read block 0x0
[148588.610379] SQUASHFS error: unable to read squashfs_super_block
[148591.633507] Buffer I/O error on dev loop0, logical block 3838912, async page read
[148594.645310] Buffer I/O error on dev loop0, logical block 3838913, async page read
[148597.656953] Buffer I/O error on dev loop0, logical block 3838914, async page read
[148600.668721] Buffer I/O error on dev loop0, logical block 3838915, async page read
[150750.559087] SQUASHFS error: squashfs_read_data failed to read block 0xea5028b7
[150750.559094] SQUASHFS error: unable to read id index table
[150750.581932] Buffer I/O error on dev loop0, logical block 3838912, async page read
[150750.593338] Buffer I/O error on dev loop0, logical block 3838913, async page read
[150750.604718] Buffer I/O error on dev loop0, logical block 3838914, async page read
[150750.616205] Buffer I/O error on dev loop0, logical block 3838915, async page read
[152505.400577] SQUASHFS error: squashfs_read_data failed to read block 0xea5028b7
[152505.400587] SQUASHFS error: unable to read id index table
[152505.423183] Buffer I/O error on dev loop0, logical block 3838912, async page read
[152505.434425] Buffer I/O error on dev loop0, logical block 3838913, async page read
[152505.445759] Buffer I/O error on dev loop0, logical block 3838914, async page read
[152505.457151] Buffer I/O error on dev loop0, logical block 3838915, async page read
[228133.182188] SQUASHFS error: squashfs_read_data failed to read block 0xea5028b7
[228133.182194] SQUASHFS error: unable to read id index table
[228133.204960] Buffer I/O error on dev loop0, logical block 3838912, async page read
[228133.216481] Buffer I/O error on dev loop0, logical block 3838913, async page read
[228133.228401] Buffer I/O error on dev loop0, logical block 3838914, async page read
[228133.239931] Buffer I/O error on dev loop0, logical block 3838915, async page read
[229734.512106] SQUASHFS error: squashfs_read_data failed to read block 0xc73e73e8
[229734.512115] SQUASHFS error: unable to read xattr id index table
[229734.534978] Buffer I/O error on dev loop0, logical block 3264320, async page read
[229734.546447] Buffer I/O error on dev loop0, logical block 3264321, async page read
[229734.557909] Buffer I/O error on dev loop0, logical block 3264322, async page read
[229734.569330] Buffer I/O error on dev loop0, logical block 3264323, async page read
[241292.999288] afs: Waiting for busy volume 536977671 (por.p.maxwell) in cell enea.it
[248447.556782] afs: Waiting for busy volume 536977671 (por.p.maxwell) in cell enea.it
[251358.922862] perf: interrupt took too long (2501 > 2500), lowering kernel.perf_event_max_sample_rate to 79000
[257073.004594] SQUASHFS error: squashfs_read_data failed to read block 0xc73e73e8
[257073.004601] SQUASHFS error: unable to read xattr id index table
[257073.027064] Buffer I/O error on dev loop0, logical block 3264320, async page read
[257073.038242] Buffer I/O error on dev loop0, logical block 3264321, async page read
[257073.049434] Buffer I/O error on dev loop0, logical block 3264322, async page read
[257073.060682] Buffer I/O error on dev loop0, logical block 3264323, async page read
[257094.784007] SQUASHFS error: squashfs_read_data failed to read block 0xea5028b7
[257094.784017] SQUASHFS error: unable to read id index table
[257100.807455] Buffer I/O error on dev loop0, logical block 3838912, async page read
```

3. I can mount the squashfs filesystem as a root without any problem, starting from a directory in AFS

On the second point I am working and I will give you a feedback soon.
Thanks,
Angelo

**Comment by dtrudg** [comment]
I did check here with the latest SingularityCE RC and stable OpenAFS and I don't have an issue (see below).

There's not anything so far that suggests a Singularity issue, as opposed to an AFS issue or bad interaction between AFS/SquashFS.

I also have the `/afs/randomroad/test` acl set to only allow `system:authuser`, to ensure there isn't anything odd going on around the privileged mount operation out of the SIF on afs. 

```
[dtrudg@dev-afs ~]$ cat /etc/redhat-release 
CentOS Linux release 7.9.2009 (Core)

[dtrudg@dev-afs ~]$ uname -a
Linux dev-afs.home.randomroad.net 3.10.0-1160.el7.x86_64 #1 SMP Mon Oct 19 16:18:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

[dtrudg@dev-afs ~]$ fs version
openafs 1.8.8

[dtrudg@dev-afs ~]$ singularity --version
singularity-ce version 3.9.0-rc.3

[dtrudg@dev-afs ~]$ singularity shell /afs/randomroad.net/test/ubuntu_latest.sif 
Singularity> echo "It works"
It works
Singularity> 
exit
```

**Comment by angelinux** [comment]
I solved my problem by building with the 3.8.4 version. The original .simg was built with singularity 3.5.2.
Thanks,
Angelo
